### DISCUSSION TITLE
How to increase the timeout for onBeforeRenderHooks

### DISCUSSION THREAD

**Discussion by snake-py** [vikejs/vike#1367] [Discussion] [answered]
We build a static web app, therefore within the onBeforeRenderHook we are doing some API calls, would it be possible to increase during build the timeout for the hook?

Currently we are getting sometimes:
```
The onBeforeRender() hook defined by /src/pages/contract/contract.page.tsx > `export { onBeforeRender }` is taking more than 4 seconds
```

**Comment by brillout** [comment, marked as answer]
@phonzammi recently implemented https://vike.dev/hooksTimeout

**Reply by brillout** [reply]
> contract.page.tsx 

You'll [need to update to the V1 design](https://github.com/vikejs/vike/commit/fe7417689cf127b6a40d7d2614a7c5b9cdc2fae4) to be able to use that config.

**Reply by snake-py** [reply]
@brillout thanks and thank you @phonzammi

**Reply by brillout** [reply]
@snake-py Btw. any news regarding sponsoring?
